### PR TITLE
feature/8582-theo-AnalyticsForRQRetries

### PR DESCRIPTION
**Pull request by theodur**
## Description of Change

Adds a new analytic event `vama_react_query_retry`, for logging how often query retries are successful. I am using the contact information query for this experiment. This event will only be logged if the contact information query has failed, and a retry occurred (which should always happen on a failure since I set the retry option to `1` for that query). It will log whether the retry was `successful` or a `fail`.

## Screenshots/Video

N/A

## Testing

Tested locally by updating the client API function to fail on the first fetch.

- [ ] Tested on iOS 
- [ ] Tested on Android 

## PR Checklist

  **Reviewer:** Confirm the items below as you review

- [ ] PR is connected to issue(s)
- [ ] Tests are included to cover this change (when possible)
- [ ] No magic strings (All string unions follow the [Union -> Constant](https://github.com/department-of-veterans-affairs/va-mobile-app/blob/develop/VAMobile/src/constants/common.ts) type pattern)
- [ ] No secrets or API keys are checked in
- [ ] All imports are absolute (no relative imports)
- [ ] New functions and Redux work have proper TSDoc annotations

## For QA

[Run a build for this branch](https://github.com/department-of-veterans-affairs/va-mobile-app/actions/workflows/on_demand_build.yml)

### CODE REVIEW

File: VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/ContactInformationScreen.tsx [hunk 1]
```diff
@@ -132,9 +132,22 @@ function ContactInformationScreen({ navigation }: ContactInformationScreenProps)
...
+
+  useEffect(() => {
+    if (failureCount > 0) {
+      setRetried(true)
+    }
+
+    if (retried && !loadingContactInformation) {
+      const retryStatus = contactInformationError ? 'fail' : 'success'
+      logAnalyticsEvent(Events.vama_react_query_retry(retryStatus))
+    }
```
Comment (dumathane) [inline VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/ContactInformationScreen.tsx:149]: Is there a reason we couldn't just do if `(failureCount > 0 && !loadingContactInformation)` ?

**Comment by dumathane** [comment]
What steps did you take to test this, Theo? Can you detail them.

**Comment by dumathane** [comment]
Tested locally by updating the client API function to fail on the first fetch.

I see this, just interested what this looked like you for you specifically in testing.

File: VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/ContactInformationScreen.tsx (hunk 1 above)
Comment (theodur) [reply VAMobile/src/screens/HomeScreen/ProfileScreen/ContactInformationScreen/ContactInformationScreen.tsx:149]: Good question. That's what I wanted to do initially, but after a retry succeeds, the failure count gets wiped, so it wouldn't tell is if the query was retried when logging successful retries

**Comment by theodur** [comment]
> Tested locally by updating the client API function to fail on the first fetch.
> 
> I see this, just interested what this looked like you for you specifically in testing.

@dumathane I modified the `get` function to fail the first time it receives the contact info endpoint

```ts

let succeed = false

export const get = async function <T>(
  endpoint: string,
  params: Params = {},
  abortSignal?: AbortSignal,
): Promise<T | undefined> {
  if (endpoint === '/v0/user/contact-info' && !succeed) {
    succeed = true
    throw new Error('fail')
  }
  return call<T>('GET', endpoint, params, undefined, abortSignal)
}

**Review by dumathane — APPROVED** [review]

**Comment by TKDickson** [comment]
Get Contact Info is called in letters, too -- on the screen that shows the mailing address where the letter is being sent. Can you make sure that we log the retries analytic there too?

**Review by TKDickson — APPROVED** [review]
approved to merge to develop

**Pull request merged**